rskj-core/src/main/java/co/rsk/peg/federation/FederationTracker.java

```diff
@@ -0,0 +1,25 @@
+package co.rsk.peg.federation;
+
+public abstract class FederationTracker<T> {
+    private T federation;
+    private boolean modified = false;
+    public boolean isModified() {
+        return this.modified;
+    }
+
+    public T getFederation() {
+        return this.federation;
+    }
+    public boolean isFederationPresent() {
+        return this.federation != null;
+    }
+
+    public void setFederationWithTracking(T aFederation) {
```

Comment:
```suggestion
    public void set(T aFederation) {
```

i find the other name a bit redundant, wdyt?

Reply:
I wanted to clarify somehow that "modified" is being set, and differentiate it from "replace"

Reply:
`setNew` ? wdyt?